bagitobjecttransfer/recordtransfer/models.py

```diff
@@ -39,138 +44,388 @@ def get_full_name(self):
+            raise ValueError(
+                f"Cannot get permanent uploaded files from session {self.token} because the "
+                "session has expired or has been deleted"
+            )
+        elif self.status in (
+            self.SessionStatus.COPYING_IN_PROGRESS,
+            self.SessionStatus.REMOVING_IN_PROGRESS,
+        ):
+            raise ValueError(
+                f"Cannot get permanent uploaded files from session {self.token} while copy or "
+                "removal of files is in progress"
+            )
+        return [f for f in self.permuploadedfile_set.all() if f.exists]
+
+    def get_uploads(self) -> Union[list[TempUploadedFile], list[PermUploadedFile]]:
+        """Get a list of temporary or permanent uploaded files associated with this session. Will
+        return temporary files if in the UPLOADING state, and permanent files if in the STORED
+        state.
+        """
+        if self.status == self.SessionStatus.CREATED:
+            return []
+        elif self.status == self.SessionStatus.UPLOADING:
+            return self.get_temporary_uploads()
+        elif self.status == self.SessionStatus.STORED:
+            return self.get_permanent_uploads()
         else:
-            logger.info("There are no existing uploaded files in the session %s", self.token)
+            raise ValueError(
+                f"Cannot get uploaded files from session {self.token} because the session status "
+                f"is {self.status} and not {self.SessionStatus.UPLOADING} or "
+                f"{self.SessionStatus.STORED}"
+            )
 
-    def move_uploads_to_permanent_storage(self, logger: Optional[logging.Logger] = None) -> None:
-        """Move uploaded files from temporary to permanent storage.
+    def remove_uploads(self) -> None:
+        """Remove all uploaded files associated with this session."""
+        if self.status == self.SessionStatus.REMOVING_IN_PROGRESS:
+            LOGGER.warning("File removal is already in progress for session %s", self.token)
+            return
+        elif self.status == self.SessionStatus.CREATED:
+            LOGGER.warning("There are no uploaded files in the session %s to remove", self.token)
+            return
+        elif self.status in (self.SessionStatus.EXPIRED, self.SessionStatus.DELETED):
+            raise ValueError(
+                f"Cannot remove uploaded files from session {self.token} because the session has "
+                "expired or has been deleted"
+            )
+        elif self.status == self.SessionStatus.COPYING_IN_PROGRESS:
+            raise ValueError(
+                f"Cannot remove uploaded files from session {self.token} while copying files is "
+                "in progress"
+            )
+
+        self.status = self.SessionStatus.REMOVING_IN_PROGRESS
+        self.save()
+
+        files = list(chain(self.permuploadedfile_set.all(), self.tempuploadedfile_set.all()))
+        for f in files:
+            f.remove()
```

Comment:
Calling `list()` here is redundant since `chain` already produces an iterator.

```suggestion
        for f in chain(self.permuploadedfile_set.all(), self.tempuploadedfile_set.all()):
            f.remove()
```